### DISCUSSION TITLE
Copilot Adapter

### DISCUSSION THREAD

**Discussion by carlosflorencio** [olimorris/codecompanion.nvim#129] [General]
Hi,

Just found this plugin and it seems very well documented and designed in general, good work :).

I'm wondering if it's possible to implement an adapter to use copilot as the backend. This would allow this plugin to replace https://github.com/CopilotC-Nvim/CopilotChat.nvim. 

avante.nvim has support for copilot [here](https://github.com/yetone/avante.nvim/blob/main/lua/avante/providers/copilot.lua).

**Comment by olimorris** [comment]
It would most definitely be possible and I've written a guide on how to do this, [here](https://github.com/olimorris/codecompanion.nvim/blob/main/doc/ADAPTERS.md). I've just been waiting for someone to do it 😉.

It is something that's been on my radar for a while but it's lower down on the priority list as I want to add some cool features like inline transformations.

If you wanted to start a PR, I'd be happy to support it and help develop it with you.

**Reply by olimorris** [reply]
@carlosflorencio as per v1.4.0 released earlier, Copilot is now available. https://github.com/olimorris/codecompanion.nvim/discussions/136

**Reply by carlosflorencio** [reply]
Wow, that was incredibly fast! I'll test it out later tonight. Thanks! 💪

**Reply by olimorris** [reply]
I could reuse most of the OpenAI adapter and the bits which required the fetching of the tokens I could borrow from [avante](https://github.com/yetone/avante.nvim).

Let me know how you get on.

**Reply by carlosflorencio** [reply]
Did a quick test and works as expected :). I'll use it more in the coming days, replacing my usage of CopilotChat.nvim. Thanks again.
